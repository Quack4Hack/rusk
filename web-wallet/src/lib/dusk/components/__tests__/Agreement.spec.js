import {
	afterEach,
	describe,
	expect,
	it
} from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { Agreement } from "..";

describe("Agreement", () => {
	const baseProps = {
		name: "test",
		controlId: "test",
		label: "I agree"
	};

	afterEach(cleanup);

	it("renders the Agreement component", () => {
		const { container } = render(Agreement, { props: { ...baseProps } });

		expect(container.firstChild).toMatchSnapshot();
	});

	// Rest of the functionality is covered under the Checkbox component tests
});
