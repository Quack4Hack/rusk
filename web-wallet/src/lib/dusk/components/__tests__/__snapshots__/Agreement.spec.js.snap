// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Agreement > renders the Agreement component 1`] = `
<div>
  <div
    class="dusk-agreement"
  >
    <input
      class="dusk-checkbox"
      id="test"
      name="test"
      tabindex="0"
      type="checkbox"
    />
     
    <label
      class="dusk-agreement__label"
      for="test"
    >
      I agree
    </label>
  </div>
</div>
`;
